### ISSUE TITLE
Attribute paths are not checked for absence of generic arguments

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#43424]
E.g. this successfully compiles on stable 1.19 / beta 1.20 /nightly
```rust
#![allow(unused)]

macro_rules! m {
    ($attr_path: path) => {
        #[$attr_path]
        fn f() {}
    }
}

m!(inline<'nonexistent_lifetime, u8, NonExistentType>);

fn main() {}
```

There are too many kinds of attributes - built in, custom, derives, legacy derives, procedural macros, legacy procedural macros, maybe something else - I'm not sure which of them are affected. Many of these attributes are removed from AST during expansion, so simply adding a check into AST validation pass won't be enough, the checking needs to be done during expansion, like for macro paths (https://github.com/rust-lang/rust/blob/3cf2c04c799024bbec8e7b7187c2e8d39ace77b6/src/librustc_resolve/macros.rs#L391).

cc @jseyfried

**Comment by petrochenkov** [comment]
This is a regression from Rust 1.18, before that paths weren't accepted in attributes.

**Comment by Mark-Simulacrum** [comment]
Nominating for prioritization.

**Comment by alexcrichton** [comment]
cc @rust-lang/compiler

**Comment by jseyfried** [comment]
Sorry for delay; still working on this.

**Comment by jseyfried** [comment]
Fixed in #43948.
